pkg/settings/setting.go

```diff
@@ -143,19 +143,36 @@ func NewSetting(name, def string) Setting {
 	return s
 }
 
+func GetEnvKey(key string) string {
+	return "CATTLE_" + strings.ToUpper(strings.Replace(key, "-", "_", -1))
+}
+
 func getMetadataConfig() string {
+	key := GetEnvKey("server-version")
+	rancherVersion := os.Getenv(key)
+	branch := "dev"
+	if releaseServerVersion(rancherVersion) {
+		branch = "master"
+	}
 	data := map[string]interface{}{
 		"url":                      "https://github.com/rancher/kontainer-driver-metadata.git",
-		"branch":                   "master",
+		"branch":                   branch,
 		"refresh-interval-minutes": "1440",
 	}
 	ans, err := json.Marshal(data)
-	if err == nil {
-		return string(ans)
+	if err != nil {
+		return ""
 	}
-	return ""
+	return string(ans)
 }
 
-func GetEnvKey(key string) string {
-	return "CATTLE_" + strings.ToUpper(strings.Replace(key, "-", "_", -1))
+func releaseServerVersion(serverVersion string) bool {
+	if serverVersion == "dev" ||
+		serverVersion == "master" ||
+		serverVersion == "" ||
+		strings.HasSuffix(serverVersion, "-head") ||
+		strings.Contains(serverVersion, "-rc") {
+		return false
```

Comment:
do we have any release versions like "alpha" / "beta"?

Reply:
updated logic to calculate server version

Reply:
does this account for `v2.4.0-alpha1` as that's what the alpha versions which are used for preview will be tagged.